src/App.tsx

```diff
@@ -1,61 +1,110 @@
 import './App.scss';
+import React, { useState } from 'react';
+import { Todo, User } from './react-app-env';
+import { TodoList } from './components/TodoList';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+function getUser(userId: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  return foundUser || null;
+}
+
+export const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
 
 export const App = () => {
+  const [todosWithUser, setTodosWithUser] = useState(todos);
+  const [todoTitle, setTodoTitle] = useState('');
+  const [selectedUserById, setSelectedUserById] = useState(0);
+  const [hasTitleError, setHasTitleError] = useState(false);
+  const [hasUserIdError, setHasUserIdError] = useState(false);
+
+  const resetForm = () => {
+    setTodoTitle('');
+    setSelectedUserById(0);
+  };
+
+  const handleFormSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    setHasTitleError(!todoTitle);
+    setHasUserIdError(!selectedUserById);
+
+    const user = getUser(selectedUserById);
+
+    const newId = Math.max(...todosWithUser.map(todo => todo.id)) + 1;
+
+    const newTodo: Todo = {
+      id: newId,
+      title: todoTitle,
+      completed: false,
+      userId: selectedUserById,
+      user,
+    };
+
+    if (newTodo && selectedUserById && todoTitle) {
```

Comment:
selectedUserById && todoTitle we can create newTodo only if we have appropriate data

in this block we create it and transfer to state